### PR TITLE
Added support to deployment manifests

### PR DESCRIPTION
**Pull request by ghost**
In order to support deployments at scale, we need to account for multiple deployments happening at any time, worker contention (on the CI engine side) and pipelines running with different versions.

To control this asyncronous flow, we need to keep some semblance of state, in the form of a file that, if it's passed to the worker, is enough information to make a deployment happen. That deployment must use the same version of the applications that previous environments have used. We can't have pipelines deploying different versions mid execution because another deployment happened in the meantime.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -8,6 +8,16 @@
 Visit the `project repository <https://github.com/OutSystems/outsystems-pipeline>`_ on GitHub for instructions on how to build example OutSystems CI/CD pipelines with common DevOps automation tools, as well as documentation that will help you adapt the examples to your particular scenarios.
 
 
+What's new
+==========
+
+* **Deployment Manifest**
```
Comment (rrmendes) [inline setup.py]: Remove bullets in subsections.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,29 @@
...
+
+Added new functions to support Air Gap deployment scenarios: 
+* deploy_apps_to_target_env_with_airgap - Deploy OutSystems Applications to a target environment without using the Deployment API
+* tag_apps_based_on_manifest_data - Syncronize LifeTime application versions from a Manifest file to target LifeTime
+
+### New Jenkins Templates
+
+New jenkins templates that include:
+* Lockable resources - Each stage has a semaphore to guarantee that the target environment does not have any ongoing deployment
+* Milestones - Previous builds will be discarded when reached a given pipeline milestone
```
Comment (rrmendes) [inline CHANGELOG.md]: Ongoing builds of the same pipeline will be discarded whenever a newer build reaches a given pipeline milestone

File: setup.py (hunk 1 above)
Comment (duarte-castano) [reply setup.py]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (duarte-castano) [reply CHANGELOG.md]: Done

**Pull request merged**